src/librustc_lint/builtin.rs

```diff
@@ -1168,18 +1168,38 @@ fn check_const(cx: &LateContext<'_, '_>, body_id: hir::BodyId) {
     let _ = cx.tcx.const_eval_poly(def_id);
 }
 
+fn check_assoc_const(cx: &LateContext<'_, '_>, body_id: hir::BodyId) {
+    let tcx = cx.tcx;
+    let def_id = tcx.hir().body_owner_def_id(body_id);
+    // Force optimized mir to be loaded, so we get lints even if the associated const is unused.
+    let _mir = tcx.optimized_mir(def_id);
```

Comment:
Didn't we agree that this part makes no difference?

Reply:
It depends on whether or not we want to lint unused associated consts in executables, per the question I asked in  https://github.com/rust-lang/rust/pull/70566#discussion_r401605179. 

If not, I'll drop this change and convert lints-used-unused to test lib vs executable lints and assert we don't emit the lint for executables.

Reply:
Doesn't https://github.com/rust-lang/rust/pull/70696 establish that we don't need this extra `optimized_mir` call to cover unused associated consts?

Reply:
It establishes it for libraries - - that test case is `#[crate_type = "lib"]`. That other `optimized_mir` call doesn't happen for executables.

Reply:
Oh I see.

However, given that (according to @oli-obk ), we don't even fully type-check dead code in binaries -- I am not sure if not linting there is truly a bug.

Reply:
Yeah, fair enough. I'll revert this change.